zallet/src/commands/migrate_zcash_conf.rs

```diff
@@ -0,0 +1,644 @@
+        .chain(Action::ignore("regtestwalletsetbestchaineveryblock"))
+        // Unsupported in `zcashd` since 5.5.0.
+        .chain(Action::ignore("sendfreetransactions"))
+        // Unsupported in `zcashd` since 5.5.0.
+        .chain(Action::ignore("txconfirmtarget"));
+
+    // Node options used directly by wallet code.
+    let node_options_direct_wallet = iter::empty()
+        // Experimental feature we aren't migrating.
+        .chain(Action::ignore("developerencryptwallet"))
+        // Used to re-enable CPU mining after disabling it during proving.
+        // Irrelevant for Zallet which doesn't include mining.
+        .chain(Action::ignore("genproclimit"))
+        // TODO: Figure out where this was used, and if we want to keep it.
+        .chain(Action::ignore("maxtxfee"))
+        // Used to check whether the configured miner address exists in the wallet.
+        // Irrelevant for Zallet which doesn't include mining.
+        .chain(Action::ignore("mineraddress"))
+        // Experimental feature we aren't migrating.
+        .chain(Action::ignore("paymentdisclosure"))
+        .chain(Some((
+            "preferredtxversion",
+            Action::warn(|value| {
+                Some(fl!(
+                    "migrate-warn-unsupported",
+                    option = "preferredtxversion",
+                    value = value,
+                ))
+            }),
+        )));
+
+    // Node options used indirectly by the `zcashd` wallet (such as in common ambient
+    // infrastructure that is being replicated in Zallet).
+    let node_options_indirect_wallet = iter::empty()
+        // This is likely the file we're migrating from; we don't want its name or path.
+        .chain(Action::ignore("conf"))
+        .chain(Some((
+            "daemon",
+            Action::warn(|value| {
+                (value != "0").then(|| fl!("migrate-warn-daemon", option = "daemon"))
+            }),
+        )))
+        // We don't want to store Zallet data in the same folder as `zcashd` data.
+        .chain(Action::ignore("datadir"))
+        // The logging systems of `zcashd` and Zallet differ sufficiently that we don't
+        // try to map existing log targets across.
+        .chain(Action::ignore("debug"))
+        // We aren't going to migrate over the `zcashd` wallet's experimental features
+        // compatibly. If we add a similar framework to Zallet it will be for from-scratch
+        // features.
+        .chain(Action::ignore("experimentalfeatures"))
+        .chain(Action::map(
+            "exportdir",
+            |config| &mut config.export_dir,
+            |value| Ok(value.into()),
+        ))
+        // TODO
+        .chain(Action::ignore("regtest"))
+        // TODO: Support mapping multi-arg options.
+        .chain(Action::ignore("rpcallowip"))
+        // Unsupported in `zcashd` since 1.0.0-beta1.
+        .chain(Action::ignore("rpcasyncthreads"))
```

Comment:
Oh, *that's* part of the cause of the excessive complexity in the zcashd RPC code. (It was a good decision to disable this, mind.)